repose-aggregator/components/filters/herp-filter/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala

```diff
@@ -121,8 +123,17 @@ class HerpFilter @Inject()(configurationService: ConfigurationService,
       else stripHeaderParams(allProjectIds.maxBy(getQuality))
     }
 
-    val projectIds = httpServletRequest.getHeaders(OpenStackServiceHeader.TENANT_ID.toString).asScala
-      .++(httpServletRequest.getHeaders(X_PROJECT_ID).asScala).toTraversable
+    def getSplitHeaders(headerInteractor: HeaderInteractor, headers: List[String]): Traversable[String] = {
+      headers.foldLeft(Traversable[String]()) { (accumulator, current) =>
+        accumulator.++(headerInteractor.getSplittableHeaders(current).asScala)
+      }
+    }
+
+    val tenantProjectHeaders = List(OpenStackServiceHeader.TENANT_ID.toString, X_PROJECT_ID)
+    val projectIds = getSplitHeaders(new HttpServletRequestWrapper(httpServletRequest), tenantProjectHeaders) match {
+      case x if x.nonEmpty => x
+      case _ => getSplitHeaders(httpServletResponse, tenantProjectHeaders)
+    }
```

Comment:
The `if`/`else` way causes the storage of a `value` whereas the `match`/`case` doesn't. This is probably a very moot point and is optimized appropriately by the compiler anyway and the first value would be as transient as the `x` variable. I don't feel strongly either way on this one.
Fixed.